apps/src/netsim/NetSimMessage.js

```diff
@@ -92,28 +92,72 @@ var NetSimMessage = module.exports = function (shard, messageRow) {
 NetSimMessage.inherits(NetSimEntity);
 
 /**
+ * @typedef {Object} MessageData
+ * @property {!number} fromNodeID - sender node ID
+ * @property {!number} toNodeID - destination node ID
+ * @property {!number} simulatedBy - node ID of client simulating message
+ * @property {*} payload - message content
+ * @property {number} extraHopsRemaining
+ * @property {number[]} visitedNodeIDs
+ */
+
+/**
+ * @typedef {Object} MessageRow
+ * @property {number} fromNodeID - this message in-flight-from node
+ * @property {number} toNodeID - this message in-flight-to node
+ * @property {number} simulatedBy - Node ID of the client responsible for
+ *           all operations involving this message.
+ * @property {Base64Payload} base64Payload - base64-encoded binary
+ *           message content, all of which can be exposed to the
+ *           student.  May contain headers of its own.
+ */
```

Comment:
Shouldn't `extraHopsRemaining` and `visitedNodeIDs` be documented in this type too?